src/components/ApplicationGroup/Details/EnvCDDetails/EnvCDDetails.tsx

```diff
@@ -0,0 +1,186 @@
+
+    const [fullScreenView, setFullScreenView] = useState<boolean>(false)
+    const [loading, result, error] = useAsync(
+        () =>
+            Promise.allSettled([
+                getCDConfig(envId),
+                getModuleConfigured(ModuleNameMap.BLOB_STORAGE),
+            ]),
+        [appId],
+    )
+    const [loadingDeploymentHistory, deploymentHistoryResult, deploymentHistoryError, , , dependencyState] = useAsync(
+        () => getTriggerHistory(+appId, +envId, pipelineId, pagination),
+        [pagination, appId, envId],
+        !!pipelineId,
+    )
+    
+    const { path } = useRouteMatch()
+    const { replace } = useHistory()
+    useInterval(pollHistory, 30000)
+    const [deploymentHistoryList, setDeploymentHistoryList] = useState<DeploymentTemplateList[]>()
+
+    useEffect(() => {
+        // check for more
+        if (loading || !deploymentHistoryResult) return
+        if (deploymentHistoryResult?.result?.length !== pagination.size) {
+            setHasMore(false)
+        } else {
+            setHasMore(true)
+            setHasMoreLoading(true)
+        }
+        const newTriggerHistory = (deploymentHistoryResult?.result || []).reduce((agg, curr) => {
+            agg.set(curr.id, curr)
+            return agg
+        }, triggerHistory)
+        setTriggerHistory(new Map(newTriggerHistory))
+    }, [deploymentHistoryResult, loading])
+
+    useEffect(() => {
+      return () => {
+          setTriggerHistory(new Map())
+          setHasMoreLoading(false)
+      }
+  }, [envId])
+
+    async function pollHistory() {
+        // polling
+        if (!pipelineId || !envId) return
+        const [error, result] = await asyncWrap(
+            getTriggerHistory(+appId, +envId, pipelineId, { offset: 0, size: pagination.offset + pagination.size }),
+        )
+        if (error) {
+            showError(error)
+            return
+        }
+
+        const triggerHistoryMap = mapByKey(result?.result || [], 'id')
+        const newTriggerHistory = Array.from(triggerHistoryMap).reduce((agg, [triggerId, curr]) => {
+            const detailedTriggerHistory = triggerHistory.has(triggerId) ? triggerHistory.get(triggerId) : {}
+            agg.set(curr.id, { ...detailedTriggerHistory, ...curr })
+            return agg
+        }, triggerHistoryMap)
+        setTriggerHistory(newTriggerHistory)
+    }
+
+    function syncState(triggerId: number, triggerDetail: History) {
+        if (triggerId === triggerDetail.id) {
+            setTriggerHistory((triggerHistory) => {
+                triggerHistory.set(triggerId, triggerDetail)
+                return new Map(triggerHistory)
+            })
+        }
+    }
+
+    if ((!hasMoreLoading && loading) || (loadingDeploymentHistory && triggerHistory.size === 0)) {
+        return <Progressing pageLoader />
+    } else if (
+        result &&
+        !Array.isArray(result[0]?.['value']?.result.pipelines)
```

Comment:
should we add optional check on result as well. please make the same changes on ci and cd details as well

Reply:
We should also add a check of pipelines length